File: macos/Scanner/Sources/Scanner/Rules/AESAUTH002.swift
public struct AESAUTH002: Rule {
    public let id = "AES-AUTH-002"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        var findings: [Finding] = []

        for server in facts.mcpServers where !server.disabled {
            for value in server.args where SecretPatterns.containsSecret(value) {
                findings.append(
                    finding(
                        app: server.appId,
                        serverName: server.name,
                        affectedPath: server.configPath,
                        maskedValue: SecretPatterns.masked(value)
                    )
                )
            }
        }

        return findings
    }
}

File: windows/Scanner/Rules/AESAUTH002.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESAUTH002 : IRule
{
    public string Id => "AES-AUTH-002";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .SelectMany(server => server.Args
                .Where(SecretPatterns.ContainsSecret)
                .Select(value => this.Finding(
                    server.AppId,
                    server.Name,
                    affectedPath: server.ConfigPath,
                    maskedValue: SecretPatterns.Masked(value)
                )))
            .ToArray();
}
